### PR TITLE
Reports-Category,designation,qualification

### PR DESCRIPTION
**Pull request by Anjanabr99**
Reports-Category,designation,qualification

### CODE REVIEW

File: src/views/Component/CustomReportMain.jsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+    const dispatch = useDispatch()
+    const onExportClick = () => {
+        if (tableDataMain.length === 0) {
+            warningNofity("Please Click The Search Button After Selecting the Options")
+        }
+        dispatch({ type: Actiontypes.FETCH_CHANGE_STATE, aggridstate: 1 })
+    }
+    const CloseReport = () => {
+        dispatch({ type: Actiontypes.FETCH_CHANGE_STATE, aggridstate: 0 })
```
Comment (itmedicity) [inline src/views/Component/CustomReportMain.jsx:47]: use it in component unmount method in useEffect

File: src/views/HrReports/QualificationReport/QualificationReport.jsx [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        },
+    ])
+    /** Course end */
+    console.log(coursslno);
+    /** Specilization start */
+    /** to get stored specilization values from redux */
+    useEffect(() => {
+        if (thirdmenu === 1) {
+            dispatch(setSpecialization(coursslno));
+        }
```
Comment (itmedicity) [inline src/views/HrReports/QualificationReport/QualificationReport.jsx:126]: else condition

File: src/views/Component/CustomAgGridMenuSelection.jsx [hunk 3]
```diff
@@ -37,14 +41,12 @@ const CustomAgGridMenuSelection = ({ tableData, columnDefs, onSelectionChanged }
             <Paper elevation={0}>
                 <Box
                     className="ag-theme-material ListItemScrol"
-                    sx={{
-                        height: { xs: 540, sm: 540, md: 540, lg: 514, xl: 802 },
-                        width: '100%',
-                    }}
+                    sx={sx}
```
[the remark this answers is not in the document]
Comment (Anjanabr99) [reply src/views/Component/CustomAgGridMenuSelection.jsx:44]: done

File: src/views/Component/CustomAgGridMenuSelection.jsx [hunk 4]
```diff
@@ -60,4 +62,4 @@ const CustomAgGridMenuSelection = ({ tableData, columnDefs, onSelectionChanged }
     )
 }
```
[the remark this answers is not in the document]
Comment (Anjanabr99) [reply src/views/Component/CustomAgGridMenuSelection.jsx:64]: done

File: src/views/Component/CustomReport.jsx [hunk 5]
```diff
@@ -10,33 +10,23 @@ import CustomAgGridMenuSelection from './CustomAgGridMenuSelection'
...
-        }
-    }, [tableDataMain.length])
 
-    /** To close the report page and back to the report list */
-    const CloseReport = async () => {
+    const onExportClick = () => {
+        dispatch({ type: Actiontypes.FETCH_CHANGE_STATE, aggridstate: 1 })
+    }
+    const CloseReport = () => {
         dispatch({ type: Actiontypes.FETCH_CHANGE_STATE, aggridstate: 0 })
```
[the remark this answers is not in the document]
Comment (Anjanabr99) [reply src/views/Component/CustomReport.jsx:26]: done

File: src/views/HrReports/CategoryReport/CategoryReports.jsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+import { ToastContainer } from 'react-toastify';
+
+
+const CategoryReports = () => {
+
+    /** to get stored category values from redux */
+    const dispatch = useDispatch();
+    useEffect(() => {
+        dispatch(setCategory());
+    }, [])
```
[the remark this answers is not in the document]
Comment (Anjanabr99) [reply src/views/HrReports/CategoryReport/CategoryReports.jsx:23]: done

File: src/views/HrReports/DesignationReport/DesignationReport.jsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+                /** to display right side table */
+                columnDefMain={columnDefMain}
+                tableDataMain={TableData}
+            />
+        </Fragment>
+    )
+}
+
+export default DesignationReport
```
[the remark this answers is not in the document]
Comment (Anjanabr99) [reply src/views/HrReports/DesignationReport/DesignationReport.jsx:138]: designation report is main file

File: src/views/HrReports/QualificationReport/QualificationReport.jsx (hunk 2 above)
Comment (Anjanabr99) [reply src/views/HrReports/QualificationReport/QualificationReport.jsx:126]: done

**Pull request merged**